docs/design/Versions and Releases.md

```diff
@@ -1,51 +1,107 @@
+the interface probably have to do changes to their code, or changes of the usage.
 
 ## Minor Version
 
-Changes in the minor version number indicate internal changes in the library, that need not be visible to an external user in any way other than the reported test results being more correct. Small interface changes are permitted, as long as they are of such a nature that they do not affect current users. That is, the interface after a minor version upgrade should be a strict superset of the interface as it was before. Adding new result messages is explicitly permitted at this level, and external code using Zonemaster must always be prepared to handle messages it has not seen before.
+Changes in the minor version number indicate that there are functional changes that might change outward-facing interface,
+but not in a breaking fashion. A user of the interface should inform themselves on what the changes are, but does not have
+to make any modifications to their own code, unless the user wants to utilize updates that require changes to the code.
+
+## Patch Version
+
+Changes in the patch version number indicate internal changes in the library, that need not be visible to an external user
+in any way other than the reported test results being more correct. Small interface changes are permitted, as long as they
+are of such a nature that they do not affect current users. That is, the interface after a patch version upgrade should be
+a strict superset of the interface as it was before. Adding new result messages is explicitly permitted at this level, and
+external code using Zonemaster must always be prepared to handle messages it has not seen before.
 
 ## Development Releases
 
-Interim releases for development use (that is, intended for developers working on the versioned component itself rather than developers using the component) will use the CPAN convention of appending an underscore followed by a number. Perl's own version parsing and handling routines will treat a version on this form as comparing in between the version number without the underscore part and the version number before it. That is, all versions of the form 1.0.1\_nn compare as being between 1.0.1 and 1.0.0. Or, to look at it slightly differently, all 1.0.1\_nn versions are steps towards what is going to be 1.0.1.
+Interim releases for development use (that is, intended for developers working on the versioned component itself rather 
+than developers using the component) will use the CPAN convention of appending an underscore followed by a number. Perl's
+own version parsing and handling routines will treat a version on this form as comparing in between the version number
+without the underscore part and the version number before it. That is, all versions of the form 1.0.1\_nn compare as being
+between 1.0.1 and 1.0.0. Or, to look at it slightly differently, all 1.0.1\_nn versions are steps towards what is going to
+be 1.0.1.
```

Comment:
If we publish Zonemaster-CLI-3.0.3_01 to CPAN, and someone runs `cpanm Zonemaster::CLI`, will they get version 3.0.2 or 3.0.3_01? If it's the latter we should probably just tag them in git, put release notes in Changes and on Github and have our bleeding edge users build their own distribution tarballs from that. If we're not to use CPAN for this purpose, we might as well follow the semver way for pre-release versions.

If we stick with the CPAN scheme, maybe put the number comparison discussion in a block quote and prefixed with "Note:"?

Reply:
Removed. We do not publish pre-releases on CPAN.